frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -822,10 +810,9 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
     end
 end
 
-function ReaderHighlight:updateHighlight(page, index, side, direction, move_by_char)
-    if self.ui.paging then return end
-    local highlight = self.view.highlight.saved[page][index]
-    local highlight_time = highlight.datetime
+function ReaderHighlight:updateHighlight(index, side, direction, move_by_char)
+    local highlight = self.ui.annotation.annotations[index]
```

Comment:
And this in a ReaderAnnotation:getAnnotationByIndex(index) :) (ok ok ok, one liners...)